src/coreclr/src/vm/object.inl

```diff
@@ -183,7 +183,7 @@ inline /* static */ TypeHandle ArrayBase::GetTypeHandle(MethodTable * pMT)
     CorElementType kind = pMT->GetInternalCorElementType();
```

Comment:
Can this just return the pMT now? It should not need to go through the type loader.

Reply:
Indeed, can just return MT.

Reply:
In fact the whole `ArrayBase::GetTypeHandle` deal is unnecessary. Same implementation as `Object`

Reply:
`Object::GetTypeHandle()` is much simpler now. `GetTrueTypeHandle` is gone. `ArrayBase::GetTypeHandle` is gone too.